packages/hardhat-core/test/internal/hardhat-network/provider/node.ts

```diff
@@ -670,8 +660,8 @@ describe("HardhatNode", () => {
 
         // We do some manual comparisons here to understand why the root of the receipt tries differ.
         if (localReceiptRoot !== remoteReceiptRoot) {
-          for (let i = 0; i < block.transactions.length; i++) {
-            const tx = block.transactions[i];
+          for (let i = 0; i < newBlock.transactions.length; i++) {
+            const tx = newBlock.transactions[i];
```

Comment:
I think the previous version was wrong, we should use `newBlock` here (it wasn't a big issue since this is only used when the receipts root don't match)